### DISCUSSION TITLE
Any general tips to convert ANNs to SNNs?

### DISCUSSION THREAD

**Discussion by sungjae-cho** [jeshraghian/snntorch#167] [General]
I am developing an SNN model to substitute ANNs with SNNs. I have observed that the performance of SNNs cannot catch up that of ANNs. To make SNNs comparable to ANNs, I tried the following:

1. Use more units in each layers to ensure high activity that possibly occurs at ANNs.
2. Use learning rates smaller than those used for ANNs.

I found that these a bit contribute to the performance of SNNs.

Do you have any other tips to convert ANNs to SNNs when it comes to setting hyperparameters?

**Comment by jeshraghian** [comment]
Converting to a discussion.

To be honest, I have not been able to find any consistent rules.
My preferred approach is to just do a large hyperparameter sweep using RayTune or Optuna, or another hyperparam search package.

If you want to convert a pre-trained ANN into an SNN, there are other packages that enable direct ANN-SNN conversion for non-sequential tasks.
